### ISSUE TITLE
Gitlab CI timeout in "next lint" with Next.js 11.1.0

### ISSUE THREAD

**Issue by ateufel** [vercel/next.js#28194]
### What version of Next.js are you using?

11.1.0

### What version of Node.js are you using?

14.x

### What browser are you using?

Chrome

### What operating system are you using?

Linux

### How are you deploying your application?

Gitlab CI

### Describe the Bug

I am using Gitlab CI to test and deploy a Next.js App, there is a Pipeline Stage for linting:
```
test:lint:
    stage: test
    dependencies:
        - prepare
    script:
        - yarn lint
```
It is pretty simple, the "prepare" Stage is bascially "yarn install" and works perfectly fine. Since Next.js 11.1.0, i get a timeout after 1h in the Gitlab Pipeline right after yarn lint:

```
$ yarn lint
info  - Loaded env from /builds/vetdoc/webapp/.env
✔ No ESLint warnings or errors
ERROR: Job failed: execution took longer than 1h0m0s seconds
```

The docker image is just a simple node 14 alpine image, and it does work perfectly fine with Next.js 11.0.1 (tried several times, it is always reproducable for me)

### Expected Behavior

The Gitlab CI Pipeline works perfectly fine with Next.js 11.1.0+

### To Reproduce

* Use Gitlab CI with a Node.js 14 Alpine Image
* Add a stage with "yarn lint" (which basically uses next lint)

**Comment by rtritto** [comment]
Same problem here.
It was introduced from [v11.0.2-canary.22](https://github.com/vercel/next.js/releases/tag/v11.0.2-canary.22) of Next.js.
Currently with [v11.1.1-canary.10](https://github.com/vercel/next.js/releases/tag/v11.1.1-canary.10), it persists.

**Comment by housseindjirdeh** [comment]
@ateufel @rtritto Would it be possible for you to share your `.gitlab-ci.yml` configuration? Couldn't reproduce it on the latest version of stable:

<img width="331" alt="Screen Shot 2021-08-18 at 11 23 16 AM" src="https://user-images.githubusercontent.com/12476932/129928117-d85f126e-20da-4cea-b8bb-9551f90704ac.png">

> It was introduced from v11.0.2-canary.22 of Next.js.

The only significant change that I noticed on v11.0.2-canary.22 was when I [changed `next lint` to be an async function](https://github.com/vercel/next.js/compare/v11.0.2-canary.21...v11.0.2-canary.22#diff-14d870abad761e1d36cbfd875474ffd988f452e3328e8ca41e58c219fc22bd5dR36). Do you think that might be the reason? (does gitlab CI stall with async functions that have no return statement?)

**Comment by rtritto** [comment]
@housseindjirdeh thanks for answer.

After some checks, in my case, the problem of unclosed process is caused by a `require('next-pwa')` in `next.config.js` file.

## Steps to reproduce
1. `yarn init -y`
2. `yarn set version berry`
3. `yarn add next react react-dom next-pwa`
4. `yarn add -D eslint eslint-config-next`
5. Create file `next.config.js`

```js
require('next-pwa')

module.exports = {}
```

6. Create file `.eslintrc.yml`

```yml
extends:
  next
```

7. Create folder `pages`
8. `yarn next lint`

**Comment by housseindjirdeh** [comment]
@rtritto Thank you so much for the detailed reproduction steps and your thoughts :). I was able to reproduce it!

> Yes, the problem should be generated adding async with this change:

Yep I was thinking it had to be that, and actually didn't even realize the `next lint` process wasn't exiting for successful outputs. Thanks for the suggestions, just put up #28299 to fix this.

**Comment by housseindjirdeh** [comment]
Thanks for logging this folks, fixed in [v11.1.1-canary.13](https://github.com/vercel/next.js/releases/tag/v11.1.1-canary.13) 🚀

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
